scan.go

```diff
@@ -221,12 +242,29 @@ func (s *ScanService) Do() (*ScanCursor, error) {
 		path += "/" + strings.Join(typesPart, ",")
 	}
 
+	// Define the body of the request
+	var body map[string]interface{}
+
+	if s.source != nil {
+		body = s.source
+	} else {
+		src, err := s.searchSource.Source()
+
+		if err != nil {
+			return nil, err
+		}
+
+		// searchSource.Source() actually returns a map, yet its return type
+		// is specified as interface...
```

Comment:
It's true, I'm relying on the internals of the function, which creates and returns a map (the (type) of that interface). The fact that you define it as an interface says nothing to the user as to the purpose of the function. It returns data, I should be able to understand what data it returns, and use that however I want, or the data shouldn't be returned (imo).

If the intention is not to make the output directly usable, I suggest not making the return value public